Adafruit_MQTT.cpp

```diff
@@ -709,8 +709,8 @@ uint16_t Adafruit_MQTT::publishPacket(uint8_t *packet, const char *topic,
     p[1] = packet_id_counter & 0xFF;
     p += 2;
 
-    // increment the packet id
-    packet_id_counter++;
+    // increment the packet id, skipping 0
+    packet_id_counter = packet_id_counter + 1 + (packet_id_counter + 1 == 0);
```

Comment:
Perform this within the check for QOS<1 on L729 instead

Reply:
@brentru This is already in a check [L707](https://github.com/adafruit/Adafruit_MQTT_Library/blob/7b2061ad85e9ba322048fb53a5e30324bd88334f/Adafruit_MQTT.cpp#L707) for QoS > 0 (and for QoS = 0 the counter is not changed and not used)